### DISCUSSION TITLE
Multiple packageRules

### DISCUSSION THREAD

**Discussion by XFNeo** [renovatebot/renovate#13664] [Request Help]
Hi! I have config like this:
```
{
    "labels": ["Dependency Bot"],
    "repositories": [
        ${REPOSITORIES}
    ],
    "extends": ["config:base", ":prHourlyLimitNone"],
    "packageRules": [
      { 
        "matchUpdateTypes": ["minor", "patch", "pin", "digest"], "automerge": true
      },
      {
        "matchPackagePatterns": ["*"],
        "allowedVersions": "!/redhat|composite|groovyless|jboss|atlassian|preview/"
      },
      {
        "matchPackageNames": ["org.jboss.arquillian.junit:arquillian-junit-standalone", "io.undertow:undertow-core"],
        "enabled": false
      }
    ]
}
```
And today I have found these MR:
commons-io:commons-io patch 2.11.0 -> 2.11.0.redhat-00001
commons-collections:commons-collections test patch 3.2.2 -> 3.2.2.redhat-2

But I have excluded any packages with word "redhat" in version in my config.
I think the problem is in  "extends": ["config:base"] because it have "workarounds:all"  -> "workarounds:mavenCommonsAncientVersion"

I can't understand how to merge this rules.

**Comment by rarkins** [comment]
It could possibly be caused by ordering of packageRules. If you enable `printConfig:true` in your config you should get a full dump of the resolved config, including `packageRules` from `config:base`. You can then confirm if your rules come before or after the one you're trying to override

**Comment by XFNeo** [comment]
Yes. it is going after my rules:
```
{
....
    "packageRules": [
      { 
        "matchUpdateTypes": ["minor", "patch", "pin", "digest"], "automerge": true
      },
      {
        "matchPackagePatterns": ["*"],
        "allowedVersions": "!/redhat|composite|groovyless|jboss|atlassian|preview/"
      },
      {
        "matchPackageNames": ["org.jboss.arquillian.junit:arquillian-junit-standalone", "io.undertow:undertow-core"],
        "enabled": false
      },
...
     {
       "matchDatasources": ["maven", "sbt-package"],
       "matchPackagePrefixes": ["commons-"],
       "allowedVersions": "!/^200\\d{5}(\\.\\d+)?/"
      },
....
    ]
}
```

**Reply by rarkins** [reply]
I think we should probably fix that ordering. rules from presets should come before any rules you define manually in the repo. If you have time to create a simpler reproduction, can you raise a bug report issue for this?

**Reply by XFNeo** [reply]
I have created issue https://github.com/renovatebot/renovate/issues/13670

**Reply by HenningCash** [reply]
We have the same problem (also with commons-io `redhat` versions). Where did #13670 go?

**Reply by rarkins** [reply]
Invalid bug reports are converted to discussions and deleted

**Reply by mikaello** [reply]
Issue 13670 was converted to this discussion: https://github.com/renovatebot/renovate/discussions/13682

... also struggling with the `redhat` versions of commons-io
